event_response_tool/layer_loss_duplicator/duplicate_layer_loss.py

```diff
@@ -0,0 +1,348 @@
+    find_column,
+    join,
+    read_byte_stream_into_csv,
+)
+
+
+logger = logging.getLogger()
+
+
+class LayerLossDuplicator:
+    def __init__(
+        self,
+        output_dir,
+        event_weights_df,
+        config,
+        layer_ids_csv=None,
+        portfolio_uuid=None,
+    ):
+        self.output_dir = output_dir
+        self.event_weights_df = event_weights_df
+        self.layer_ids_csv = layer_ids_csv
+        self.portfolio_uuid = portfolio_uuid
+        self.config = config
+
+        self.layer_list = []  # List of LayerViews
+        self.loss_set_mapping = {}  # Old LossSet UUID : New LossSet UUID
+
+    def extract_layers(self):
+        if not self.layer_ids_csv is None and len(str(self.layer_ids_csv)) > 0:
+            # get the list of LayerViews from the input CSV
+            layer_list_df = read_input_file(self.layer_ids_csv)
+            layer_column = find_column(
+                "layer", layer_list_df.columns.tolist()
+            )
+            self.layer_list = layer_list_df[layer_column].unique().tolist()
+        else:
+            if not self.portfolio_uuid is None:
+                # get the list of LayerViews from the PortfolioView UUID
+                try:
+                    are_portfolio = analyzere.PortfolioView.retrieve(
+                        self.portfolio_uuid
+                    )
+                except Exception as e:
+                    alert.exception(
+                        f"Could not retrieve PortfolioView {self.portfolio_uuid}: {e}"
+                    )
+                else:
+                    self.layer_list = [
+                        layer_view.id
+                        for layer_view in are_portfolio.layer_views
+                    ]
+                    alert.info(
+                        f"Fetched {len(self.layer_list)} LayerView from PortfolioView {self.portfolio_uuid}"
+                    )
+
+    # Generic Layer to replace filter layer in structure
+    def replace_filter_layer(self, loss_sets):
+        unlimited = sys.float_info.max
+        currency = self.config.get("defaults", "currency")
+        generic_layer = analyzere.Layer(
+            type="Generic",
+            description="Generic Layer",
+            loss_sets=[loss_sets],
+            participation=1,
+            aggregate_attachment=MonetaryUnit(0, currency),
+            aggregate_limit=MonetaryUnit(unlimited, currency),
+            attachment=MonetaryUnit(0, currency),
+            franchise=MonetaryUnit(0, currency),
+            limit=MonetaryUnit(unlimited, currency),
+        ).save()
+        return generic_layer
+
+    def scale_elt(self, elt_df, loss_set):
+        alert.debug(f"Scaling loss_set {loss_set.id}")
+        # Normalize the EventId column
+        event_id_column = find_column("event", elt_df.columns.tolist())
+        self.event_weights_df = self.event_weights_df.rename(
+            columns={self.event_weights_df.columns[0]: event_id_column}
```

Comment:
Is it worth making this more generic to handle users uploading an event weights csv with the column order `Weight,Event`?